### DISCUSSION TITLE
EU VAT Laws and cash collection based on VAT numbers

### DISCUSSION THREAD

**Discussion by inpresif** [thedevdojo/wave#176] [Q&A]
How do you verify VAT numbers for B2B orders in the EU and collect the funds accordingly?
Any smart suggestions i.c.w. Wave?

**Comment by bobbyiliev** [comment]
Hey!

That should be handled on the payment provider that you are using. Is it Stripe or Paddle in your case?

**Reply by inpresif** [reply]
HI Bobby, we'll use Stripe. Thank you.

**Reply by bobbyiliev** [reply]
Hey,

Wave utilizes [Stripe Checkout](https://docs.stripe.com/payments/checkout) as per the docs [here](https://devdojo.com/wave/docs/features/billing#stripe).

You can collect customer tax IDs with Checkout as per the documentation here:

> https://docs.stripe.com/tax/checkout/tax-ids

Hope that this helps!

**Reply by inpresif** [reply]
I'll check it out, thank you.

**Reply by bobbyiliev** [reply]
No problem, good luck with your project.

**Reply by inpresif** [reply]
I'd like to circle back on this 1 sec ;-)

We do not see any tax on our product (in sandbox mode). We seem to have it set up correctly in Stripe. Any idea where I can add this to Wave/Laravel?

![image](https://github.com/user-attachments/assets/5744e854-6354-4ed2-90a8-d7502c47850d)

**Reply by inpresif** [reply]
I added something here

`app\wave\src\Http\Livewire\Billing\Checkout.php`

```
'automatic_tax' => ['enabled' => true], // PP: Enable automatic tax calculation
'tax_id_collection' => ['enabled' => true], // Collect VAT number
```

and that does work on the checkout page. I assume/know I must add this in some other areas as well, but I think this should be a feature in your settings to be honest, juyst because this will be overwritten on an update :)

```
    public function redirectToStripeCheckout(Plan $plan)
    {
        $stripe = new StripeClient(config('wave.stripe.secret_key'));

        $price_id = $this->billing_cycle_selected == 'month' ? $plan->monthly_price_id : $plan->yearly_price_id ?? null;

        $checkout_session = $stripe->checkout->sessions->create([
            'line_items' => [[
                'price' => $price_id,
                'quantity' => 1
            ]],
            'metadata' => [
                'billable_type' => 'user',
                'billable_id' => auth()->user()->id,
                'plan_id' => $plan->id,
                'billing_cycle' => $this->billing_cycle_selected
            ],
            'mode' => 'subscription',
            'automatic_tax' => ['enabled' => true], // PP: Enable automatic tax calculation
            'tax_id_collection' => ['enabled' => true], // Collect VAT number
            'success_url' => url('subscription/welcome'),
            'cancel_url' => url('settings/subscription'),
        ]);

        return redirect($checkout_session->url);
    }
```
